ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/homescreenpage/HomeScreenContainerFragmentTest.java

```diff
@@ -0,0 +1,206 @@
+
+        //Act
+        final View viewLocal = homeScreenContainerFragment.onCreateView(layoutInflater, viewGroup, bundle);
+
+        //Assert
+        errorCollector.checkThat(viewLocal, is(view));
+    }
+
+    @Test
+    public void onActivityCreated() {
+        //Arrange
+        Whitebox.setInternalState(homeScreenContainerFragment, "mPresenter", presenter);
+
+        //Act
+        homeScreenContainerFragment.onActivityCreated(bundle);
+
+        //Assert
+        verify(presenter, times(1)).initPage();
+    }
+
+    @Test
+    public void showAndHideConversationButton() throws NoSuchMethodException {
+        //Arrange
+        hasPageLoaded();
+        Whitebox.setInternalState(homeScreenContainerFragment, "mRoot", view);
+        when(view.findViewById(R.id.ko__new_conversation_button)).thenReturn(view);
+
+        //Act
+        homeScreenContainerFragment.showNewConversationButton();
+        homeScreenContainerFragment.hideNewConversationButton();
+
+        //Assert
+        verify(view, times(2)).setVisibility(integerArgumentCaptor.capture());
+        errorCollector.checkThat(integerArgumentCaptor.getAllValues().get(0), is(View.VISIBLE));
+        errorCollector.checkThat(integerArgumentCaptor.getAllValues().get(1), is(View.GONE));
+    }
+
+    @Test
+    public void onViewCreated() throws NoSuchMethodException {
+        //Arrange
+        hasPageLoaded();
+        Whitebox.setInternalState(homeScreenContainerFragment, "mPresenter", presenter);
+        Whitebox.setInternalState(homeScreenContainerFragment, "mRoot", view);
+        when(view.findViewById(R.id.ko__new_conversation_button)).thenReturn(view);
+        when(fragmentManager.findFragmentById(R.id.ko__home_screen_list_fragment)).thenReturn(homeScreenListFragment);
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                Object[] arguments = invocation.getArguments();
+                View.OnClickListener onClickListener = (View.OnClickListener)arguments[0];
+                onClickListener.onClick(view);
+                return null;
+            }
+        }).when(view).setOnClickListener(any(View.OnClickListener.class));
+
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                Object[] arguments = invocation.getArguments();
+                HomeScreenListContract.OnScrollListListener onScrollListListener =
+                        (HomeScreenListContract.OnScrollListListener)arguments[0];
+                onScrollListListener.onScroll(true);
+                return null;
+            }
+        }).when(homeScreenListFragment).setOnScrollListener(any(HomeScreenListContract.OnScrollListListener.class));
+
+        //Act
+        homeScreenContainerFragment.onViewCreated(view, bundle);
+        verify(presenter).onScrollList(booleanArgumentCaptor.capture());
```

Comment:
❗️ This snippet of code mixes Act and Assert phases. Since this is against AAA guideline and "Split Personality" test smell ([https://tinyurl.com/ybxbgp79](https://tinyurl.com/ybxbgp79)) and severely affects readability, it is recommended to follow the example and establish AAA order.  

**[More details](https://confluence.devfactory.com/x/iQCIEw)**
          
## The solution in this case

**Note** When there are dependencies between phases and mixing is required then this is a clear sign that the test should be split into multiple tests.  
```Java  
// Act
homeScreenContainerFragment.onViewCreated(view, bundle); 

// Assert
verify(presenter).onScrollList(booleanArgumentCaptor.capture());
```
 

<a id="8dbc6e50-54f4-11e8-9db9-b7d507f482ce" href="none" codereview-uuid-reserved-tag />